### Reading, writing and displaying images with OpenCV

#### Loading the first image

In [7]:
# import openCV
import cv2 

# Load an image using 'imread' specifying the path to image
input = cv2.imread('./images/elephant.jpg')

# Our file 'elephant.jpg' is now loaded and stored in python 
# as a varaible we named 'input'

# To display our image variable, we use 'imshow'
# The first parameter will be title shown on image window
# The second parameter is the image varialbe
cv2.imshow('Elephant''s image', input)

# 'waitKey' allows us to input information when a image window is open
# By leaving it blank it just waits for anykey to be pressed before 
# continuing. By placing numbers (except 0), we can specify a delay for
# how long you keep the window open (time is in milliseconds here)
cv2.waitKey()

# This closes all open windows 
# Failure to place this will cause your program to hang
cv2.destroyAllWindows()

#### Storing images

In [8]:
print(input.shape)

(519, 778, 3)


#### Shape gives the dimensions of the image array

The 2D dimensions are 519 pixels in high bv 778 pixels wide.
The '3L' means that there are 3 other components (RGB) that make up this image.

In [9]:
# print dimensions of the image

print('Height of Image:', int(input.shape[0]), 'pixels')
print('Width of Image: ', int(input.shape[1]), 'pixels')

Height of Image: 519 pixels
Width of Image:  778 pixels


#### Save images edited in OpenCV

In [10]:
# using 'imwrite'
cv2.imwrite('el_output.jpg', input)
cv2.imwrite('el_output.png', input)

True